cypress/integration/version/task_filters.ts

```diff
@@ -218,31 +225,64 @@ describe("Tasks filters", () => {
     });
 
     it("Clicking on 'All' checkbox adds all the base statuses and clicking again removes them", () => {
-      cy.toggleTableFilter(3);
       const taskStatuses = ["All", "Succeeded"];
-      cy.getInputByLabel("All").check({ force: true });
-      taskStatuses.forEach((status) => {
-        cy.getInputByLabel(status).should("be.checked");
-      });
-
-      cy.toggleTableFilter(3);
+      selectCheckboxOption("All");
+      assertChecked(taskStatuses, true);
       urlSearchParamsAreUpdated({
         pathname: pathTasks,
         paramName: urlParam,
-        search: "all,success",
-      });
-
-      cy.getInputByLabel("All").uncheck({ force: true });
-      taskStatuses.forEach((status) => {
-        cy.getInputByLabel(status).should("not.be.checked");
+        search: "all",
       });
-      cy.toggleTableFilter(3);
+      waitForTable();
 
+      selectCheckboxOption("All");
+      assertChecked(taskStatuses, false);
       urlSearchParamsAreUpdated({
         pathname: pathTasks,
         paramName: urlParam,
-        search: "",
+        search: null,
       });
     });
   });
 });
+
+/**
+ * Function used to assert that table exists and is not loading.
+ */
+const waitForTable = () => {
+  cy.dataCy("tasks-table").should("be.visible");
+  cy.dataCy("tasks-table").should("not.have.attr", "data-loading", "true");
+};
+
+/**
+ * Function used to assert if checkboxes with certain labels are checked or unchecked.
+ * @param statuses list of labels to assert on
+ * @param checked true if should be checked, false if should be unchecked
+ */
+const assertChecked = (statuses: string[], checked: boolean) => {
+  cy.get(":not(.ant-dropdown-hidden) > .ant-table-filter-dropdown")
+    .find(".cy-checkbox")
+    .each((el) => {
+      expect(statuses).to.include(el.text());
+      if (checked) {
+        cy.wrap(el).find('input[type="checkbox"]').should("be.checked");
+      } else {
+        cy.wrap(el).find('input[type="checkbox"]').should("not.be.checked");
+      }
+    });
+};
+
+/**
+ * Function used to select a checkbox option from the table filter dropdown.
+ * @param label label of the checkbox option to click on
+ */
+const selectCheckboxOption = (label: string) => {
+  cy.get(":not(.ant-dropdown-hidden) > .ant-table-filter-dropdown")
+    .find(".cy-checkbox")
+    .should("not.be.disabled")
+    .each((el) => {
+      if (el.text() === label) {
+        cy.wrap(el).click({ scrollBehavior: false });
```

Comment:
Does `cy.getInputByLabel()` work? It should return the checkbox that corresponds to the label.
cypress provides a [`.check`](https://docs.cypress.io/api/commands/check) method for interacting with checkbox and safely checking them.